app/page/home/activity.tsx

```diff
@@ -51,181 +64,159 @@ const UserOpHistoryItem: React.FC<{
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              1.2
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">ETH</div>
-          </div>
-          {/* To address */}
-          <div className="leading-[19px] text-[16px] text-[#bbbbbb] whitespace-nowrap">
-            to: 0x0000...00000
-          </div>
-        </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
-          </div>
-        </div>
-      </a>
-    )
+  const tokenInfo: TokenTransferInfo = {
+    symbol: `${chainName}ETH`,
+    value,
+    to
   }
 
-  if (txLog.status === TransactionStatus.Failed) {
-    const userOpHash = txLog.receipt.userOpHash
-    return (
-      <a
-        className="w-full flex items-center p-[14px_0px_14px_0px]"
-        href={`${explorerUrl}userOpHash/${userOpHash}?network=${chainName}`}
-        target="_blank">
-        {/* Status and to address */}
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and token amount */}
-          <div className="flex items-center mb-[4px]">
-            <OctagonXmark className="w-[16px] h-[16px] mr-[8px]" />
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              Failed
-            </div>
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              1.2
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">ETH</div>
-          </div>
-          {/* To address */}
-          <div className="leading-[19px] text-[16px] text-[#bbbbbb] whitespace-nowrap">
-            to: 0x0000...00000
-          </div>
-        </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
-          </div>
-        </div>
-      </a>
-    )
+  // Check if the transaction "to" address is a stored token
+  const tokenStored = tokens.find((token) => address.isEqual(token.address, to))
+
+  // If a token is found, consider it an ERC20 token transfer
+  if (tokenStored) {
+    try {
+      const { to: tokenTo, value: tokenValue } = decodeTransferParams(data)
+      tokenInfo.symbol = tokenStored.symbol
+      tokenInfo.value = tokenValue
+      tokenInfo.to = tokenTo
+      // TODO: Handling non-transfer actions
+    } catch (e) {
+      console.warn(`[app] Account doing non-transfer actions: ${e.message}`)
```

Comment:
[future] Non transfer token interaction should be treated as contract interaction. Could be refined in later PR.

Reply:
OK!